src/citrine/informatics/modules.py

```diff
@@ -0,0 +1,26 @@
+"""Tools for working with design spaces."""
+from typing import Any, List, Mapping, Type
+from uuid import UUID
+
+from citrine._serialization.polymorphic_serializable import PolymorphicSerializable
+from citrine._serialization.serializable import Serializable
+
+
+class Module(PolymorphicSerializable['Module']):
+    """A Citrine Module - an abstract type that returns the proper
+    subtype based on the 'type' value of the passed in dict.
+    """
+
+    _response_key = None
+
+    @classmethod
+    def get_type(cls, data) -> Type['Module']:
+        """Return the subtype."""
+        from citrine.informatics.design_spaces import DesignSpace
+        from citrine.informatics.processors import Processor
+        from citrine.informatics.predictors import Predictor
```

Comment:
why import these here instead of at the top?

Reply:
Otherwise you have a circular import...There is probably a bigger conversation to be had here, but I wanted to do one thing at a time.

Reply:
ohhh. thats generally a sign that things are not organized properly. Who do the sub-types need to subclass `Module`? Cant `Module` just be alongside `Predictor`, `Processor` and `DesignSpace`?

Reply:
I think the smell here is that a super-type knows about its subtypes directly.  That arrow should be one-way.  If a super-type has to know about all types that inherit from it you are breaking a pretty big design principal of OO.

Reply:
Agreed. Playing around with it right now to figure out if there's a way around this.

Reply:
I think this is a case of trying to do two things with the same object. For inheritance, obviously the subtype needs to know about the supertype. For deserialization, I don't see how you get around the supertype knowing about the subtype. In fact that pattern is used throughout the citrine-python codebase. It's just usually in the same file, so it doesn't seem as bad.

Reply:
I think the answer is to not have the supertype responsible for deserialization.  This is sort of an anti-pattern that I see a lot in the scala codebase as well, but if we can get it to work then whatevs

Reply:
We can certainly get it to work. It works in this example. I don't see how you do deserialization without having the supertype be responsible for it - you're not going to know the subtype ahead of time.

Reply:
I have read a bunch of different serialization library's code and do not see anyone that has a solution much different than this. The "best" solution I've seen involves using decorators to add objects to a global registry. Either way, I think this is a different conversation.